mypy/checkexpr.py

```diff
@@ -71,6 +71,15 @@
 MAX_UNIONS = 5  # type: Final
 
 
+# Types considered safe for comparisons with --strict-equality due to known behaviour of __eq__.
+OVERLAPPING_TYPES_WHITELIST = [('builtins.set', 'builtins.frozenset'),
+                               ('typing.KeysView', 'builtins.set'),
+                               ('typing.KeysView', 'builtins.frozenset')]  # type: Final
```

Comment:
Is it worth doing Values also? (And Items?) (Genuinely not sure.)

Reply:
Interesting:
```
>>> {1: 2}.values() == {2}
False
>>> {1: 2}.items() == {(1, 2)}
True
```
So it works for keys and items, but not values.

Reply:
There was actually a discussion about this on the python-ideas mailing list recently. Even comparing `d.values() == d.values()` (where d is a dict) returns False.